ipa-core/src/report.rs

```diff
@@ -564,6 +899,36 @@ mod test {
         assert_eq!(dec_report, report);
     }
 
+    #[test]
+    fn enc_dec_roundtrip_oprf() {
```

Comment:
we should also add tests (prop tests maybe) that reject invalid reports

Reply:
You mean I randomly generate ciphertexts (or encrypted reports) and try do decrypt them and assert that it fails? I agree that we could concerned here that we don't implement the encryption decryption correctly. The roundtrip only checks for correctness, not for security. However, I think generating random ciphertexts and trying to decrypt them would not ensure that we encrypt/decrypt correctly.